core/lib/src/server.rs

```diff
@@ -351,28 +360,30 @@ impl Rocket {
     //
     // On catcher failure, the 500 error catcher is attempted. If _that_ fails,
     // the (infallible) default 500 error cather is used.
-    pub(crate) async fn handle_error<'s, 'r: 's>(
+    pub(crate) fn handle_error<'s, 'r: 's>(
```

Comment:
Why this change? It seems unnecessary? The code also got harder to read.

In general, these tail `instrument` calls are hard to follow.

Reply:
This could alternatively remain an `async fn` and use the `#[instrument]` attribute, like:
```rust
#[tracing::instrument(
    level = "warn",
    skip(self, status, req),
    fields(message = %format_args!("Responding with {} catcher.", Paint::red(&status)),
)]
pub(crate) async fn handle_error<'s, 'r: 's>(
    // ...
```